src/server.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use futures_util::{SinkExt, StreamExt, TryFutureExt};
+use log::{error, info};
+use tokio::sync::{mpsc, RwLock};
+use tokio_stream::wrappers::UnboundedReceiverStream;
+use warp::ws::{Message, WebSocket};
+use warp::Filter;
+
+use rtpeeker_common::packet::PacketType;
+use rtpeeker_common::Packet;
+
+use crate::sniffer::Sniffer;
+
+static DIST_DIR: &str = "client/dist";
+static WS_PATH: &str = "ws";
+static NEXT_CLIENT_ID: AtomicUsize = AtomicUsize::new(1);
+
+type Clients = Arc<RwLock<HashMap<usize, mpsc::UnboundedSender<Message>>>>;
+type Packets = Arc<RwLock<Vec<Packet>>>;
+
+pub async fn run<T: pcap::Activated + 'static>(sniffer: Sniffer<T>, addr: SocketAddr) {
+    let packets = Packets::default();
+    let clients = Clients::default();
+
+    let clients_sniff = clients.clone();
+    let packets_sniff = packets.clone();
+
+    tokio::task::spawn(async move {
+        sniff(sniffer, packets_sniff, clients_sniff).await;
+    });
+
+    let packets_filter = warp::any().map(move || packets.clone());
+    let clients_filter = warp::any().map(move || clients.clone());
+    let ws = warp::path(WS_PATH)
+        .and(warp::ws())
+        .and(packets_filter)
+        .and(clients_filter)
+        .map(|ws: warp::ws::Ws, packets, clients| {
+            ws.on_upgrade(move |socket| client_connected(socket, packets, clients))
+        });
+
+    let routes = ws.or(warp::fs::dir(DIST_DIR));
+
+    warp::serve(routes).run(addr).await;
+}
+
+async fn client_connected(ws: WebSocket, packets: Packets, clients: Clients) {
+    let client_id = NEXT_CLIENT_ID.fetch_add(1, Ordering::Relaxed);
+
+    info!("New client connected, assigned id: {}", client_id);
+
+    let (mut client_ws_tx, mut client_ws_rx) = ws.split();
```

Comment:
For me better name would be `send_channel, receive_channel`

Reply:
`tx` and `rx` are used by convention as stated in [this page in Rust book](https://doc.rust-lang.org/book/ch16-02-message-passing.html). I basically copied what in this [Warp WebSocket example](https://github.com/seanmonstar/warp/blob/master/examples/websockets_chat.rs) but maybe simpyl `ws_tx` and `ws_rs` would be better.